liana-gui/src/app/settings/mod.rs

```diff
@@ -111,6 +114,184 @@ where
     Ok(())
 }
 
+/// Cubes represent user accounts that can contain multiple features (Vault, Active wallet, etc.)
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct CubeSettings {
+    pub id: String,
+    pub name: String,
+    pub network: Network,
+    pub created_at: i64,
+    /// The Vault wallet for this Cube (optional - may not be set up yet)
+    pub vault_wallet_id: Option<WalletId>,
+    /// Optional security PIN (stored as Argon2id hash with salt in PHC format)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub security_pin_hash: Option<String>,
```

Comment:
We could use `pbkdf2` (with 500K iterations) from the `bitcoin_hashes` crate to avoid adding any new dependencies